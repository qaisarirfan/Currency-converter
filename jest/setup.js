jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');

  // The mock for `call` immediately calls the callback which is incorrect
  // So we override it with a no-op
  Reanimated.default.call = () => {};

  return Reanimated;
});

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useFocusEffect: () => jest.fn(),
    useRoute: () => jest.fn(),
    useNavigation: () => ({
      navigation: () => jest.fn(),
      canGoBack: () => jest.fn(),
      goBack: () => jest.fn(),
      push: () => jest.fn(),
    }),
  };
});

jest.mock('react-native-bootsplash', () => ({
  hide: jest.fn(),
  show: jest.fn(),
}));

jest.mock('react-native-localize', () => ({
  getLocales: () => [
    { countryCode: 'GB', languageTag: 'en-GB', languageCode: 'en', isRTL: false },
    { countryCode: 'US', languageTag: 'en-US', languageCode: 'en', isRTL: false },
    { countryCode: 'FR', languageTag: 'fr-FR', languageCode: 'fr', isRTL: false },
  ],

  getNumberFormatSettings: () => ({
    decimalSeparator: '.',
    groupingSeparator: ',',
  }),

  getCalendar: () => 'gregorian', // or "japanese", "buddhist"
  getCountry: () => 'US', // the country code you want
  getCurrencies: () => ['USD', 'EUR'], // can be empty array
  getTemperatureUnit: () => 'celsius', // or "fahrenheit"
  getTimeZone: () => 'Europe/Paris', // the timezone you want
  uses24HourClock: () => true,
  usesMetricSystem: () => true,

  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
}));

jest.mock('react-native-keyboard-aware-scroll-view', () => {
  const KeyboardAwareScrollView = require('react-native').ScrollView;
  return { KeyboardAwareScrollView };
});

jest.mock('react-native-gesture-handler', () => {
  // eslint-disable-next-line global-require
  const View = require('react-native/Libraries/Components/View/View');
  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {},
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    DrawerLayoutAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Other */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  };
});

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
