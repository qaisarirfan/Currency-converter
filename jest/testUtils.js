import React from 'react';
import { I18nextProvider } from 'react-i18next';
import { Provider } from 'react-redux';
import { render } from 'react-native-testing-library';
import configureStore from 'redux-mock-store';

import { initialState as themesState } from '../src/redux/reducers/themes';
import configureI18next from '../src/services/i18n';

const INITIAL_STATE = {
  themes: themesState,
};
const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
  render(ui, {
    wrapper: ({ children }) => (
      <Provider store={store}>
        <I18nextProvider i18n={configureI18next()}>{children}</I18nextProvider>
      </Provider>
    ),
    ...options,
  });

// re-export everything for convenience
export * from 'react-native-testing-library';

// override render method
export { reduxRender as render, store };
