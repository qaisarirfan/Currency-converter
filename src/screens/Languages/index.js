import React from 'react';
import { View, SafeAreaView, ScrollView, Image, I18nManager } from 'react-native';
import { useDispatch, useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';
import Entypo from 'react-native-vector-icons/Entypo';
import i18next from 'i18next';
import RNBootSplash from 'react-native-bootsplash';
import RNRestart from 'react-native-restart';

import RowItem, { RowSeparator } from '../../components/RowItem';
import HeaderBar from '../../components/HeaderBar';

import { createLanguageAction } from '../../redux/reducers/languages/actionCreators';
import { selectLanguage } from '../../redux/reducers/languages/selectors';
import { selectStyleableTheme } from '../../redux/reducers/themes/selectors';

import constants from '../../constants';
import themeStyles from './styles';

function Languages() {
  const { t } = useTranslation();
  const dispatch = useDispatch();

  const styleableTheme = useSelector(selectStyleableTheme);
  const currentLang = useSelector(selectLanguage);

  const styles = themeStyles(styleableTheme);

  const { allowedLanguages } = constants;
  const keys = Object.keys(allowedLanguages);

  const changeLanguage = ({ code, rtl }) => {
    i18next.changeLanguage(code).then(() => {
      dispatch(createLanguageAction(code));
      I18nManager.forceRTL(rtl);
      RNBootSplash.show({ duration: 700 });
      setTimeout(() => {
        RNRestart.Restart();
      }, 1000);
    });
  };

  return (
    <View style={styles.root}>
      <HeaderBar title={t('common.languages')} />
      <SafeAreaView>
        <ScrollView>
          {keys.map((languageCode) => {
            const language = allowedLanguages[languageCode];
            return (
              <React.Fragment key={language.code}>
                <RowItem
                  testID={language.code}
                  title={language.title}
                  onPress={() => changeLanguage(language)}
                  rightIcon={
                    <View
                      style={{
                        flexDirection: 'row',
                      }}
                    >
                      {currentLang === languageCode ? (
                        <View style={{ marginRight: 12 }}>
                          <Entypo name="check" size={20} color={styleableTheme[50]} />
                        </View>
                      ) : null}
                      <View>
                        <Image source={language.flag} />
                      </View>
                    </View>
                  }
                />
                <RowSeparator />
              </React.Fragment>
            );
          })}
        </ScrollView>
      </SafeAreaView>
    </View>
  );
}

export default Languages;
