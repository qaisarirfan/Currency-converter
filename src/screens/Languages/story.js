import React from "react"
import Languages from "."

export default {
  title: "Languages",
  storyFn() {
    return (
      <div>
        <Languages />
      </div>
    )
  },
}
