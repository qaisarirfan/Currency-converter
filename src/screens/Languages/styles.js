import { StyleSheet } from 'react-native';

const styles = (theme) =>
  StyleSheet.create({
    root: {
      backgroundColor: theme[500],
      flex: 1,
    },
  });

export default styles;
