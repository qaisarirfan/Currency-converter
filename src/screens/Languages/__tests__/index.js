import React from "react"
import {render, cleanup, store} from '../../../../jest/testUtils';
import Languages from ".."

afterEach(() => {
  cleanup();
  store.clearActions();
});

describe("<Languages />", () => {
  it("Does not explode", () => {
    const component = render(<Languages />).toJSON();
    expect(component).toMatchSnapshot()
  })
})
