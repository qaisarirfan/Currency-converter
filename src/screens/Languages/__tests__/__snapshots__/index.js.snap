// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Languages /> Does not explode 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#4f6d7a",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "backgroundColor": "#36505d",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 20,
          },
          Object {
            "backgroundColor": "#36505d",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "space-between",
            "paddingBottom": 12,
            "paddingTop": 12,
          },
          Object {
            "backgroundColor": "#36505d",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "justifyContent": "center",
            "paddingLeft": 6,
            "width": 250,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "opacity": 1,
            }
          }
          testID="back"
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#eaedef",
                  "fontSize": 16,
                },
                undefined,
                Object {
                  "fontFamily": "Entypo",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "#eaedef",
                "fontSize": 18,
                "lineHeight": 18,
              }
            }
          >
            Back
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#36505d",
            "justifyContent": "center",
            "width": 250,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#eaedef",
              "fontSize": 18,
              "lineHeight": 18,
            }
          }
        >
          Languages
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "justifyContent": "center",
            "paddingRight": 6,
            "width": 250,
          }
        }
      />
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <RCTScrollView>
      <View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#8499a2",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 16,
            }
          }
          testID="en"
        >
          <Text
            style={
              Object {
                "color": "#eaedef",
                "fontSize": 16,
              }
            }
          >
            English
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 12,
                }
              }
            >
              <Text
                allowFontScaling={false}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#eaedef",
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Entypo",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <View>
              <Image
                source={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 0.5,
            }
          }
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#8499a2",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 16,
            }
          }
          testID="ur"
        >
          <Text
            style={
              Object {
                "color": "#eaedef",
                "fontSize": 16,
              }
            }
          >
            اردو
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View>
              <Image
                source={
                  Object {
                    "process": [Function],
                  }
                }
              />
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "height": 0.5,
            }
          }
        />
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
